src/main/java/com/example/hbs/service/HotelService.java

```diff
@@ -109,7 +119,9 @@ public HotelResponseDto updateHotel(Long id, HotelResponseDto hotelResponseDto,
         Optional.ofNullable(hotel).orElseThrow(() -> new HbsException("Hotel not found"));
         checkIfUserIsOwner(hotel, userId);
         if (hotelResponseDto.getAvailableRooms() != null) {
+            Integer roomsAdded = hotelResponseDto.getAvailableRooms() - hotel.getAvailableRooms();
             hotel.setAvailableRooms(hotelResponseDto.getAvailableRooms());
+            hotel.setNoOfRooms(hotel.getNoOfRooms() + roomsAdded);
```

Comment:
logic?

Reply:
If we are adding or subtracting Available rooms without adding or deleting bookings So no of rooms should also be increased/decreased accordingly, right?